### PR TITLE
Ingress controller SSL unit tests

### PR DESCRIPTION
**Pull request by ijsnellf**
Adds unit tests for the Envoy SSL config generation and does some minor refactoring to make it easier to test.

### CODE REVIEW

**Comment by ijsnellf** [comment]
Contributes to https://github.com/istio/manager/issues/278.

File: cmd/manager/main.go [hunk 1]
```diff
@@ -107,13 +107,17 @@ var (
...
 				IngressClass:    flags.ingressClass,
 			}
-			if flags.defaultIngressController {
-				controllerConfig.IngressSyncMode = kube.IngressDefault
-			}
 			controller := kube.NewController(cmd.Client, controllerConfig)
-			w, err := envoy.NewIngressWatcher(controller, controller,
-				&model.IstioRegistry{ConfigRegistry: controller},
-				cmd.Client, &flags.proxy, flags.ingressSecret, cmd.RootFlags.Namespace)
+			context := &envoy.IngressContext{
```
Comment (kyessenov) [inline cmd/manager/main.go]: I think `IngressConfig` is more appropriate here - proxy context is used to differentiate  multiple envoy instances, ingress proxies seem to act the same way. We can make a more general proxy context later that includes "ingress role" for the proxy, if you want.

File: proxy/envoy/ingress.go [hunk 2]
```diff
@@ -179,44 +185,26 @@ func generateIngress(namespace, secret string, secrets model.SecretRegistry,
 		Listeners: listeners,
 		Admin: Admin{
 			AccessLogPath: DefaultAccessLog,
-			Address:       fmt.Sprintf("tcp://0.0.0.0:%d", mesh.AdminPort),
+			Address:       fmt.Sprintf("tcp://0.0.0.0:%d", context.Mesh.AdminPort),
```
Comment (kyessenov) [inline proxy/envoy/ingress.go]: nite: there is `WildcardAddress` that can be used for `0.0.0.0`

File: cmd/manager/main.go (hunk 1 above)
Comment (ijsnellf) [inline cmd/manager/main.go]: I agree, that is better.

File: proxy/envoy/ingress.go (hunk 2 above)
Comment (ijsnellf) [inline proxy/envoy/ingress.go]: fixed

**Pull request merged**